### PR TITLE
Apr 166 find provider on register

### PR DESCRIPTION
**Pull request by David-B-Read**
Create API for searching organisations in the RoATP by UKPRN or company name

### CODE REVIEW

File: src/SFA.DAS.RoATPService.Application.Api/StartupConfiguration/Startup.cs [hunk 1]
```diff
@@ -0,0 +1,180 @@
...
+
+                if (env.IsDevelopment()) app.UseDeveloperExceptionPage();
+
+                app.UseSwagger()
+                    .UseSwaggerUI(c =>
+                    {
+                        c.SwaggerEndpoint("/swagger/v1/swagger.json", "SFA.DAS.RoATPService.Application.Api v1");
+                    });
+                //    .UseAuthentication();
+
```
Comment (MarkFCain) [inline src/SFA.DAS.RoATPService.Application.Api/StartupConfiguration/Startup.cs]: Why is this commented out? Can you check you can't leave it in?

File: src/SFA.DAS.RoATPService.Data/OrganisationSearchRepository.cs [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+        public async Task<IEnumerable<Organisation>> OrganisationSearchByName(string organisationName)
+        {
+            var connectionString = _configuration.SqlConnectionString;
+
+            using (var connection = new SqlConnection(connectionString))
+            {
+                if (connection.State != ConnectionState.Open)
+                    await connection.OpenAsync();
+
+                var organisations = await connection.QueryAsync<Organisation>($"select * from [Organisations] where LegalName LIKE '%{organisationName}%' OR TradingName LIKE '%{organisationName}%'");
```
Comment (MarkFCain) [inline src/SFA.DAS.RoATPService.Data/OrganisationSearchRepository.cs]: see my previous comment.  Inject in a variable, and add this: ",new {organisationName}"

File: src/SFA.DAS.RoATPService.Application.Api/StartupConfiguration/Startup.cs (hunk 1 above)
Comment (David-B-Read) [reply src/SFA.DAS.RoATPService.Application.Api/StartupConfiguration/Startup.cs]: Good spot, commented out in error

File: src/SFA.DAS.RoATPService.Data/OrganisationSearchRepository.cs [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+            long ukPrnValue = Convert.ToInt64(ukPrn);
+
+            var connectionString = _configuration.SqlConnectionString;
+
+            using (var connection = new SqlConnection(connectionString))
+            {
+                if (connection.State != ConnectionState.Open)
+                    await connection.OpenAsync();
+
+                var organisations = await connection.QueryAsync<Organisation>($"select * from [Organisations] where UKPRN = {ukPrnValue}");
```
[the remark this answers is not in the document]
Comment (David-B-Read) [reply src/SFA.DAS.RoATPService.Data/OrganisationSearchRepository.cs]: Done

File: src/SFA.DAS.RoATPService.Data/OrganisationSearchRepository.cs (hunk 2 above)
Comment (David-B-Read) [reply src/SFA.DAS.RoATPService.Data/OrganisationSearchRepository.cs]: Done

**Pull request merged**